starknet_py/net/models/transaction.py

```diff
@@ -1,29 +1,224 @@
+
+
+@dataclass(frozen=True)
+class DeployAccount(AccountTransaction):
+    """
+    Represents a transaction in the StarkNet network that is a deployment of a StarkNet account
+    contract.
+    """
+
+    class_hash: int = field(metadata={"marshmallow_field": Felt()})
+    contract_address_salt: int = field(metadata={"marshmallow_field": Felt()})
+    constructor_calldata: List[int] = field(
+        metadata={"marshmallow_field": fields.List(fields.String())}
+    )
+    version: int = field(metadata={"marshmallow_field": Felt()})
+
+    type: TransactionType = TransactionType.DEPLOY_ACCOUNT
+
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        """
+        Calculates the transaction hash in the StarkNet network.
+        """
+        contract_address = compute_address(
+            salt=self.contract_address_salt,
+            class_hash=self.class_hash,
+            constructor_calldata=self.constructor_calldata,
+            deployer_address=0,
+        )
+        return compute_deploy_account_transaction_hash(
+            version=self.version,
+            contract_address=contract_address,
+            class_hash=self.class_hash,
+            constructor_calldata=self.constructor_calldata,
+            max_fee=self.max_fee,
+            nonce=self.nonce,
+            salt=self.contract_address_salt,
+            chain_id=chain_id.value,
+        )
+
+
+@dataclass(frozen=True)
+class Invoke(AccountTransaction):
+    """
+    Represents a transaction in the StarkNet network that is an invocation of a Cairo contract
+    function.
+    """
+
+    contract_address: int = field(metadata={"marshmallow_field": Felt()})
+    calldata: List[int] = field(
+        metadata={"marshmallow_field": fields.List(fields.String())}
+    )
+    nonce: Optional[int] = field(metadata={"marshmallow_field": NoneFelt()})
+    # A field element that encodes the signature of the invoked function.
+    # The entry_point_selector is deprecated for version 1 and above (transactions
+    # should go through the '__execute__' entry point).
+    entry_point_selector: Optional[int] = field(
+        default=None, metadata={"marshmallow_field": NoneFelt()}
+    )
+
+    type: TransactionType = TransactionType.INVOKE
+
+    @marshmallow.post_dump
+    def _remove_entry_point_selector(
+        self, data: Dict[str, Any], many: bool, **kwargs
+    ) -> Dict[str, Any]:
+        # pylint: disable=no-self-use, unused-argument
+        data["type"] = "INVOKE_FUNCTION"
+
+        version = int(data["version"], 16)
+        if version == 0:
+            return data
+
+        assert (
+            data["entry_point_selector"] is None
+        ), f"entry_point_selector should be None in version {version}."
+        del data["entry_point_selector"]
+
+        return data
+
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        """
+        Calculates the transaction hash in the StarkNet network.
+        """
+        if self.version == 0:
```

Comment:
Maybe some tests for this would be useful so codecov stops complaining? Also for the code above and below.

Reply:
What about this?

Reply:
Will add them